### PR TITLE
Add shared preferences package

### PR DESCRIPTION
**Pull request by mxwnk**

### CODE REVIEW

**Comment by t-animal** [comment]
We should clarify if @ohaz wants to build the tauri-based desktop client before investing in desktop code.

**Comment by ohaz** [comment]
I definitely want to build a desktop client, not sure about tauri yet though.

File: frontend-mobile/macos/Flutter/GeneratedPluginRegistrant.swift [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import FlutterMacOS
 import Foundation
 
+import shared_preferences_macos
 
 func RegisterGeneratedPlugins(registry: FlutterPluginRegistry) {
+  SharedPreferencesPlugin.register(with: registry.registrar(forPlugin: "SharedPreferencesPlugin"))
```
Comment (t-animal) [inline frontend-mobile/macos/Flutter/GeneratedPluginRegistrant.swift]: Reading the docs on my cell phone I didn't see this. Why do you think it's necessary?

**Comment by mxwnk** [comment]
I would have used this package to persist the user name within NSUserDefauls / SharedPreferences on mobile. Imho this does not depend on our desktop client or did I get something wrong

File: frontend-mobile/macos/Flutter/GeneratedPluginRegistrant.swift (hunk 1 above)
Comment (t-animal) [reply frontend-mobile/macos/Flutter/GeneratedPluginRegistrant.swift]: Oh wait, this was autogenerated, because you ran it on MacOs, right?

Comment (mxwnk) [reply frontend-mobile/macos/Flutter/GeneratedPluginRegistrant.swift]: I think so yeah

**Pull request merged**